File: dotnet/src/EPostak/Resources/AccountResource.cs
using EPostak.Models;

namespace EPostak.Resources;

/// <summary>
/// Resource for retrieving account information — firm details, subscription
/// plan, and document usage for the current billing period.
/// <para>
/// For key introspection, OAuth token minting, and key rotation see
/// <see cref="AuthResource"/> on <c>client.Auth.*</c>.
/// </para>
/// </summary>
/// <example>
/// <code>
/// var account = await client.Account.GetAsync();
/// Console.WriteLine($"Plan: {account.Plan.Name} ({account.Plan.Status})");
/// Console.WriteLine($"Usage: {account.Usage.Outbound} sent, {account.Usage.Inbound} received");
/// </code>
/// </example>
public sealed class AccountResource
{
    private readonly HttpRequestor _http;

    internal AccountResource(HttpRequestor http) => _http = http;

    /// <summary>
    /// Get account information for the authenticated API key.
    /// Returns the associated firm, current subscription plan, and usage counters.
    /// </summary>
    public Task<Account> GetAsync(CancellationToken ct = default)
        => _http.RequestAsync<Account>(HttpMethod.Get, "/account", ct);

    /// <summary>Get per-firm plan and current-period license usage.</summary>
    public Task<Dictionary<string, object?>> LicenseInfoAsync(CancellationToken ct = default)
        => _http.RequestAsync<Dictionary<string, object?>>(HttpMethod.Get, "/licenses/info", ct);
}

File: java/src/main/java/sk/epostak/sdk/resources/AccountResource.java
package sk.epostak.sdk.resources;

import sk.epostak.sdk.HttpClient;
import sk.epostak.sdk.models.Account;

import java.util.Map;

/**
 * Account and firm information.
 * <p>
 * Provides access to the current account's firm details, subscription plan,
 * and document usage counters.
 * <p>
 * For key introspection, OAuth token minting, and key rotation see
 * {@code client.auth()}.
 * <p>
 * Access via {@code client.account()}.
 *
 * <pre>{@code
 * Account acct = client.account().get();
 * System.out.println("Firm: " + acct.firm().name());
 * System.out.println("Plan: " + acct.plan().name());
 * System.out.println("Outbound usage: " + acct.usage().outbound());
 * }</pre>
 */
public final class AccountResource {

    private final HttpClient http;

    /**
     * Creates a new account resource.
     *
     * @param http the HTTP client used for API communication
     */
    public AccountResource(HttpClient http) {
        this.http = http;
    }

    /**
     * Get the current account information, including firm details, subscription
     * plan, and document usage counters.
     *
     * @return the account details
     * @throws sk.epostak.sdk.EPostakException if the request fails
     */
    public Account get() {
        return http.get("/account", Account.class);
    }

    @SuppressWarnings("unchecked")
    public Map<String, Object> licenseInfo() {
        return http.get("/licenses/info", Map.class);
    }
}
